src/com/facebook/buck/apple/AppleBinaryDescription.java

```diff
@@ -178,24 +178,28 @@ public AppleBinaryDescription(
   @Override
   public boolean hasFlavors(
       ImmutableSet<Flavor> flavors, TargetConfiguration toolchainTargetConfiguration) {
-    if (FluentIterable.from(flavors).allMatch(SUPPORTED_FLAVORS::contains)) {
+    Set<Flavor> unmatchedFlavors = Sets.difference(flavors, SUPPORTED_FLAVORS);
+    if (unmatchedFlavors.isEmpty()) {
       return true;
     }
     ImmutableSet<Flavor> delegateFlavors =
-        ImmutableSet.copyOf(Sets.difference(flavors, NON_DELEGATE_FLAVORS));
-    if (swiftDelegate
-        .map(swift -> swift.hasFlavors(delegateFlavors, toolchainTargetConfiguration))
-        .orElse(false)) {
+      ImmutableSet.copyOf(Sets.difference(flavors, NON_DELEGATE_FLAVORS));
+    ImmutableSet<Flavor> supportedDelegateFlavors = swiftDelegate
+      .map(swift -> swift.getSupportedFlavors(delegateFlavors, toolchainTargetConfiguration))
+      .orElse(ImmutableSet.<Flavor>of());
+    unmatchedFlavors = Sets.difference(unmatchedFlavors, supportedDelegateFlavors);
```

Comment:
Would it be valuable to just make this an immutable set now, rather than potentially make 2 later?